File: Repositories/js/js-monolith-images/metadata.js
import { metadataIndex } from "./dataLoadModule.js";

export function getImageMetadata(ids) {
    return ids
        .map(id => metadataIndex[id])
        .filter(Boolean)
        .map(m => ({
            id: m.id,
            name: m.name,
            tags: m.tags
        }));
}

File: Repositories/python/python-monolith-images/src/metadata.py
from src.data_load_module import metadata_index


def get_image_metadata(image_ids):
    return [
        {
            "id": metadata_index[i]["id"],
            "name": metadata_index[i]["name"],
            "tags": metadata_index[i]["tags"],
        }
        for i in image_ids
        if i in metadata_index
    ]
